src/main.py

```diff
@@ -1264,9 +1253,11 @@ def _return_to_cluster(self, new_primary, role, is_dead=False):
             last_op = self.zk.noexcept_get('%s/%s/op' % (self.zk.MEMBERS_PATH, helpers.get_hostname()))
             logging.info('Last op is: %s' % str(last_op))
             if role != 'primary' and not self.is_op_destructive(last_op) and not self._is_simple_primary_switch_tried():
-                logging.info('Trying to do a simple primary switch.')
+                logging.info('Trying to do a simple WAL source switch to: {}'.format(new_primary))
                 result = self._try_simple_primary_switch_with_lock(limit, new_primary, is_dead)
-                logging.info('Primary switch count: %s finish with result: %s', self.checks['primary_switch'], result)
+                logging.info('WAL source switch count: %s finish with result: %s', self.checks['primary_switch'], result)
+                if result:
+                    self.checks['primary_switch'] = 0
```

Comment:
There is already setting `primary_switch` to 0 inside `_try_simple_primary_switch_with_lock` before returning `True` here: https://github.com/yandex/pgconsul/pull/59/files#diff-2e5ad92c43aa96cc3a9cef6c6aec998b216f1379c43b1f651013d25e55989312R1048 and here: https://github.com/yandex/pgconsul/pull/59/files#diff-2e5ad92c43aa96cc3a9cef6c6aec998b216f1379c43b1f651013d25e55989312R1053

I think, we can either add it to the last possible place when True can be returned: https://github.com/yandex/pgconsul/pull/59/files#diff-2e5ad92c43aa96cc3a9cef6c6aec998b216f1379c43b1f651013d25e55989312R1077
Or we can remove it from everywhere and set it outside `_try_simple_primary_switch_with_lock` call.

Either way is good, but please not a mix of them.